src/lib/components/Swap/SwapButton.tsx

```diff
@@ -26,17 +28,26 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
     setActiveTrade((activeTrade) => activeTrade && trade.trade)
   }, [trade])
 
-  // TODO(zzmp): Track pending approval
-  const useIsPendingApproval = () => false
-
   // TODO(zzmp): Return an optimized trade directly from useSwapInfo.
-  const optimizedTrade = useSwapApprovalOptimizedTrade(trade.trade, allowedSlippage, useIsPendingApproval)
+  const optimizedTrade =
+    // Use trade.trade if there is no swap optimized trade. This occurs if approvals are still pending.
+    useSwapApprovalOptimizedTrade(trade.trade, allowedSlippage, useIsPendingApproval) || trade.trade
```

Comment:
Couldnt this cause a case where the user submits the non optimized trade? aka shouldnt we always wait for approvals? Or is this out of scope

Reply:
trade.trade will still return the correct approvals from `useSwapApproval`, so this will cause the `approval` to be `ApprovalState.PENDING`, which is what we want. The follow-up PR will amend the UI to "wait" for approval so that a user does not submit a non-optimized trade.

Reply:
ahh i see, makes sense